common4j/src/main/com/microsoft/identity/common/java/commands/parameters/CommandParameters.java

```diff
@@ -116,6 +117,10 @@ public String getApplicationIdentifier(){
         return String.format(APPLICATION_IDENTIFIER_FORMAT, this.callerPackageName, this.callerSignature);
     }
 
+    public boolean hasNestedAppParameters() {
+        // return true even if one of the nested app params is present
```

Comment:
Will the request succeed with even one of these?

Reply:
it should not succeed. ESTS should throw an error. However, the reason why I am returning true even if one of the params are present is to specify that we found a nested app paramater and not to miss appending this parameter when doing the token request.